src/collection/Map.php

```diff
@@ -97,7 +106,7 @@ public function setAll($collection) {
 	 * Removes and returns an element from the map by the given key. Returns null if the key
 	 * does not exist.
 	 * 
-	 * @param string $key
+	 * @param string|Text $key
 	 * @return mixed the element at the given key
 	 */
 	public function remove($key) {
```

Comment:
Should we return `self` here, too (see `Set::remove`, `ArrayList::remove`)?

Reply:
Yes, very good catch. The idea of this library is to promise a consistent flow. We should stick to it and deliver like that. Very well spotted leak here.